airflow/utils/helpers.py

```diff
@@ -179,6 +182,74 @@ def f(t):
     return s
 
 
+def kill_using_shell(pid, signal=signal.SIGTERM):
+    try:
+        process = psutil.Process(pid)
+    except psutil.NoSuchProcess:
+        return
+    # Use sudo only when necessary - consider SubDagOperator and SequentialExecutor case.
+    if process.username() != getpass.getuser():
+        args = ["sudo", "kill", "-{}".format(int(signal)), str(pid)]
+    else:
+        args = ["kill", "-{}".format(int(signal)), str(pid)]
+    # PID may not exist and return a non-zero error code
+    subprocess.call(args)
+
+
+def kill_process_tree(logger, pid):
```

Comment:
Use the GPID instead of the PID, just use the kill(1) command but instead of giving it a process number, give it the negation of the group number.

 http://stackoverflow.com/questions/392022/best-way-to-kill-all-child-processes